# 20221013_TIL
## 오늘 할 일
- [X] TIL 특장
- [X] 머신러닝 3, 4주차
- [X] 타임어택

## 오늘 한일
### TIL 특강
오늘 TIL 특강을 들었다.<br>
내가 그동안 써왔던 TIL의 문제점을 알 수 있었다.<br>
여러가지가 있지만 그 중에서도 내가 알게 된 점을 구체적으로 적지 않은 것 이 가장 큰 문제점이었던 것 같다.<br>
강의를 듣고 필기한 것을 쓰는 것이 아니라, 강의를 듣고 내가 알게 된 점을 써야 한다.<br>
그래야 내가 성장한 것들 내 성향 등을 볼 수 있는 나만의 TIL이 될 수 있다.<br>
하지만 생각없이 그냥 강의를 필기한 것만 올린 것 같다.<br>
요즘엔 더 그게 심했던 것 같다.<br>
프로젝트가 끝나고 주말+공휴일이 지난 후에 다시 공부하려고 했을 때 집중이 잘 안됐다. <br>
그렇다고 주말과 공휴일에 공부를 안한 것도 아니다.<br>
공부하는 것이 재미가 없는게 아니라 손에 안 잡혔다.<br>
공부를 시작하면 재밌지만 시작하기가 어려웠다.<br>
집중을 못하는 시간이 늘면서 내가 생각해논 일정이 점점 꼬이는 것도 문제가 있었다.<br>
점점 하루에 해야 할 양이 많아지고, 마음은 급해져서 집중은 안되고 이런 것이 반복됐다.<br>
이게 그냥 컨디션이 안좋은 건 줄 알았는데, 오늘 특강을 들으면서 다시 생각해보니 몰랐던 점과 알게된 점을 적지 않아서 헤메는 것도 있었고, 뭔가 아예 새로운 것을 배운다는 느낌이 들어서 헤메는 것도 있는 것 같았다.<br>
또 구체적인 목표를 정해두지 않은 것도 있는 것 같다.<br>
취업이라는 목표가 있지만 구체적이진 않았다.<br>
어디에 취업하고 싶은지, 어떤 걸 하고 싶은지 아직 명확하지 않다.<br>
새롭게 배우는 것이 재밌어서 아직 내가 어떤 기술이 더 좋은 지, 나에게 더 잘 맞는지는 잘 모르겠다.<br>
내가 좋아하는 것은 시간이 지나가면서 점점 알게 될 것 같다.<br>
그래서 일단은 시간표에 충실해보려고 한다.<br>
하루에 할 수 있는 양이 어느정도인지 확인하고, 계획을 다시 세워야 할 것 같다.<br>
그리고 주말을 활용해서 진도를 따라 잡아야 겠다.<br>
TIL을 쓸 때는 이제 내가 다른 사람에게 이해갈 수 있게 설명이 가능한지, 내가 다시 이 문제를 만나면 당황하지 않고 풀 수 있는지를 기준으로 한 번 작성해보려고 한다.<br>
이전에 쓴 TIL을 고칠 순 없지만 지금부터 다시 시작하는 마음으로 세롭게 써봐야 겠다.<br>
<br>

### 타임어택
문제점
models.py
```python
class User(AbstractUser):
    class Meta:
        db_table = 'User'
        
    phone = models.CharField(max_length=24)
    address = models.CharField(max_length=256)
```

```
SystemCheckError: System check identified some issues:

ERRORS:
auth.User.groups: (fields.E304) Reverse accessor for 'User.groups' clashes with reverse accessor for 'User.gruops'.
    HINT: Add or change a related_name argument to the definition for 'User.groups' or 'User.gruops'.
auth.User.user_permissions: (fields.E304) Reverse accessor for 'User.user_permissions' clashes with reverse accessor for 'User.user_permissions'.
    HINT: Add or change a related_name argument to the definition for 'User.user_permissions' or 'User.user_permissions'.
user.User.groups: (fields.E304) Reverse accessor for 'User.groups' clashes with reverse accessor for 'User.gruops'.
    HINT: Add or change a related_name argument to the definition for 'User.groups' or 'User.gruops'.
user.User.user_permissions: (fields.E304) Reverse accessor for 'User.user_permissions' clashes with reverse accessor for 'User.user_permissions'.
    HINT: Add or change a related_name argument to the definition for 'User.user_permissions' or 'User.user_permissions'.
```
`models.py`를 작성하고 `makemigrations`를 하려는데 이런 error가 떴다.
이게 뭔지 몰라서 class이름을 바꿔보고, table이름도 바꿔봤는데 같은 error가 떴다.
```python
class UserModel(AbstractUser):
    class Meta:
        db_table = 'UserModel'
        
    phone = models.CharField(max_length=24)
    address = models.CharField(max_length=256)
```

해결
settings.py
```python
AUTH_USER_MODEL = 'user.User'
```
`settings.py`에 위의 코드를 넣어주니 해결이 됐다
원래 알았던 것이었는데, timeateck이라 급하기도 하고, 막상 error가 뜨니 당황해서 시간을 더 많이 썻던 것 같다.
